pipelines/controller/src/main/java/com/google/fhir/analytics/ApiController.java

```diff
@@ -166,4 +178,60 @@ public ResponseEntity<String> testView(
     }
     return new ResponseEntity<>(response, status);
   }
+
+  @GetMapping("/next")
+  public ScheduleDto getNextScheduled() {
+    ScheduleDto schedule = new ScheduleDto();
+    LocalDateTime nextRun = pipelineManager.getNextIncrementalTime();
+    if (nextRun == null) {
+      schedule.setNextRun("NOT SCHEDULED");
+    } else {
+      schedule.setNextRun(nextRun.toString());
+    }
+    return schedule;
+  }
+
+  @GetMapping("/dwh")
+  public DwhDto getDwh() {
+    DwhDto dwhDto = new DwhDto();
+    String dwh = pipelineManager.getCurrentDwhRoot();
+    dwhDto.setDwhPrefix(dataProperties.getDwhRootPrefix());
+    dwhDto.setDwhPath(dwh == null || dwh.isEmpty() ? "" : dwh);
+    dwhDto.setDwhSnapshots(dwhFilesManager.listDwhSnapshots());
+    return dwhDto;
+  }
+
+  @DeleteMapping("/dwh")
+  @ResponseStatus(HttpStatus.NO_CONTENT)
```

Comment:
Are we communicating any error messages in case of failures?

Reply:
Yeah, once an exception occurs Spring handles it as a 500 error with more details in the response object.